### PR TITLE
[SPARK-29587][DOC][FOLLOWUP] Add `SQL` tab in the `Data Types` page

### PR DESCRIPTION
**Pull request by xuanyuanking**
### What changes were proposed in this pull request?
Add the new tab `SQL` in the `Data Types` page.

### Why are the changes needed?
New type added in SPARK-29587.

### Does this PR introduce any user-facing change?
No.

### How was this patch tested?
Locally test by Jekyll.
image

### CODE REVIEW

**Comment by xuanyuanking** [comment]
retest this please.

**Comment by xuanyuanking** [comment]
cc @cloud-fan

**Comment by xuanyuanking** [comment]
retest this please.

**Comment by cloud-fan** [comment]
This is not a breaking change and we don't need to update migration guide.

We should update `sql-ref-datatypes.md`

**Comment by xuanyuanking** [comment]
@cloud-fan Thanks for pointing out, add a new section in data types page.

File: docs/sql-ref-datatypes.md [hunk 1]
```diff
@@ -631,4 +631,71 @@ from pyspark.sql.types import *
...
+  <td> <b>BinaryType</b> </td>
+  <td> string </td>
+</tr>
+<tr>
+  <td> <b>BinaryType</b> </td>
+  <td> string </td>
+</tr>
+<tr>
+  <td> <b>DecimalType</b> </td>
+  <td> decimal, desc, numeric </td>
```
Comment (dongjoon-hyun) [inline docs/sql-ref-datatypes.md]: Ur, `desc`?

File: docs/sql-ref-datatypes.md [hunk 2]
```diff
@@ -631,4 +631,71 @@ from pyspark.sql.types import *
...
+
+The following table shows the type names as well as aliases used in Spark SQL parser for each data type.
+
+<table class="table">
+<tr>
+  <th style="width:50%">Data type</th>
+  <th>SQL name</th></tr>
+<tr>
+  <td> <b>BooleanType</b> </td>
+  <td> boolean </td>
```
Comment (dongjoon-hyun) [inline docs/sql-ref-datatypes.md]: Shall we capitalize SQL types like `BOOLEAN` and `INT`?

File: docs/sql-ref-datatypes.md [hunk 3]
```diff
@@ -631,4 +631,71 @@ from pyspark.sql.types import *
...
+</tr>
+<tr>
+  <td> <b>ByteType</b> </td>
+  <td> byte, tinyint </td>
+</tr>
+<tr>
+  <td> <b>ShortType</b> </td>
+  <td> shot, smallint </td>
```
Comment (dongjoon-hyun) [inline docs/sql-ref-datatypes.md]: `shot`?

File: docs/sql-ref-datatypes.md [hunk 4]
```diff
@@ -631,4 +631,71 @@ from pyspark.sql.types import *
...
+  <td> string </td>
+</tr>
+<tr>
+  <td> <b>BinaryType</b> </td>
+  <td> string </td>
+</tr>
+<tr>
+  <td> <b>BinaryType</b> </td>
+  <td> string </td>
+</tr>
```
Comment (dongjoon-hyun) [inline docs/sql-ref-datatypes.md]: It seems to be repeated twice: `BinaryType` .

**Review by dongjoon-hyun — CHANGES_REQUESTED** [review]
Thanks, @xuanyuanking .  Please review the document once more. It seems that there exist a few typos.

File: docs/sql-ref-datatypes.md [hunk 5]
```diff
@@ -631,4 +631,67 @@ from pyspark.sql.types import *
 </tr>
 </table>
 </div>
+
+<div data-lang="sql"  markdown="1">
```
Comment (cloud-fan) [inline docs/sql-ref-datatypes.md]: nit: `SQL`

File: docs/sql-ref-datatypes.md [hunk 6]
```diff
@@ -631,4 +631,67 @@ from pyspark.sql.types import *
...
+  <td> BINARY </td>
+</tr>
+<tr>
+  <td> <b>DecimalType</b> </td>
+  <td> DECIMAL, DEC, NUMERIC </td>
+</tr>
+<tr>
+  <td> <b>CalendarIntervalType</b> </td>
+  <td> INTERVAL </td>
+</tr>
```
Comment (cloud-fan) [inline docs/sql-ref-datatypes.md]: shall we mention complext type as well?
```
StructType
STRUCT<field1_type, field2_type, ...>

ArrayType
ARRAY<element_type>

MapType
MAP<key_type, value_type>
```

File: docs/sql-ref-datatypes.md (hunk 5 above)
Comment (xuanyuanking) [reply docs/sql-ref-datatypes.md]: The `sql` is not display name here.

File: docs/sql-ref-datatypes.md (hunk 6 above)
Comment (xuanyuanking) [reply docs/sql-ref-datatypes.md]: Sorry for missing, done in d4780c6

File: docs/sql-ref-datatypes.md (hunk 1 above)
Comment (xuanyuanking) [reply docs/sql-ref-datatypes.md]: Sorry for the rush, should not push before dinner... Done in 99d0db6.

File: docs/sql-ref-datatypes.md (hunk 2 above)
Comment (xuanyuanking) [reply docs/sql-ref-datatypes.md]: Copy, change all the type name. Done in 99d0db6.

File: docs/sql-ref-datatypes.md (hunk 3 above)
Comment (xuanyuanking) [reply docs/sql-ref-datatypes.md]: Sorry, done in 99d0db6.

File: docs/sql-ref-datatypes.md (hunk 4 above)
Comment (xuanyuanking) [reply docs/sql-ref-datatypes.md]: Thanks, done in 99d0db6.

**Review by cloud-fan — APPROVED** [review]

**Review by dongjoon-hyun — APPROVED** [review]
+1, LGTM. Thank you, @xuanyuanking and @cloud-fan .
I also verified the result locally.
Merged to master/3.0.

**Comment by xuanyuanking** [comment]
Thanks for the help!

**Pull request closed without merging**